azure/functions/http.py

```diff
@@ -21,12 +21,14 @@ def __init__(self,
                  route_params: typing.Mapping[str, str],
                  body_type: str,
                  body: typing.Union[str, bytes]) -> None:
+
+        body_str: typing.Optional[str] = None
```

Comment:
Only body_str could be None.
The request should always have body bytes. It acts as the raw body of the request itself.
The body_str should be encoded with 'utf-8' directly, rather than doing lazy encoding in get_json() method.